src/web.ts

```diff
@@ -1,22 +1,61 @@
-import { WebPlugin } from '@capacitor/core';
-import { SignInWithApplePlugin } from './definitions';
+import { WebPlugin } from "@capacitor/core";
 
-export class SignInWithAppleWeb extends WebPlugin implements SignInWithApplePlugin {
+import { SignInWithApplePlugin, InitOptions } from "./definitions";
+
+declare var window: any;
+
+export class SignInWithAppleWeb extends WebPlugin
+  implements SignInWithApplePlugin {
   constructor() {
     super({
-      name: 'SignInWithApple',
-      platforms: ['web']
+      name: "SignInWithApple",
+      platforms: ["web"],
     });
   }
 
-  async Authorize(): Promise<{response: any}> {
+  async SignIn(): Promise<void> {
+    if (window && window.AppleID) {
+      return await window.AppleID.auth.signIn();
+    }
+
     return;
   }
+
+  async Init(options: InitOptions): Promise<void> {
+    this.loadAppleScript(() => {
+      if (window && window.AppleID) {
+        const { clientId, scope, redirectURI, state, usePopup } = options;
+
+        window.AppleID.auth.init({
+          clientId: clientId,
+          scope: scope,
+          redirectURI: redirectURI,
+          state: state,
+          usePopup: usePopup !== undefined ? usePopup : false,
+        });
+      }
+    });
+  }
+
+  async Authorize(): Promise<{ response: any }> {
+    return await window.AppleID.auth.signIn();
+  }
+
+  private loadAppleScript(callback: any) {
```

Comment:
feels like this could do with some error handling

Reply:
@priyankpat- this still needs some error handling, why not do something like:
```
loadAppleScript(): Promise<any> {
  return new Promise((resolve, reject) => {
    const script = document.createElement('script');

    script.src = src;
    script.onload = resolve;
    script.onerror = reject;

    document.querySelector('head').appendChild(script);
  });
}
```

So you can keep a consistent promise-based approach?